verde/utils.py

```diff
@@ -281,3 +282,64 @@ def grid_to_table(grid):
     coord_dict.update(var_dict)
     data = pd.DataFrame(coord_dict)
     return data
+
+
+def load_surfer(fname, dtype='float64'):
+    """
+    Read data from a Surfer ASCII grid file.
+
+    Surfer is a contouring, griding and surface mapping software
+    from GoldenSoftware. The names and logos for Surfer and Golden
+    Software are registered trademarks of Golden Software, Inc.
+
+    http://www.goldensoftware.com/products/surfer
+
+    This feature was ported from Fatiando, but instead of
+    output being a dictionary, in verde the output is a `xarray.DataArray`
+    that contains the easting and northing coordinates, and data values.
+
+    Parameters:
+
+    * fname : str
+        Name of the Surfer grid file
+    * dtype : numpy dtype object or string
+        The type of variable used for the data. Default is numpy.float64. Use
+        numpy.float32 if the data are large and precision is not an issue.
+
+    Returns:
+
+    * data : :class: `xarray.DataArray`
+        A 2D grid with the data variables.
+    """
+    # Surfer ASCII grid structure
+    # DSAA            Surfer ASCII GRD ID
+    # nCols nRows     number of columns and rows
+    # xMin xMax       X min max
+    # yMin yMax       Y min max
+    # zMin zMax       Z min max
+    # z11 z21 z31 ... List of Z values
+    with open(fname) as input_file:
+        # DSAA is a Surfer ASCII GRD ID (discard it for now)
+        input_file.readline()
+        # Read the number of columns (ny) and rows (nx)
+        ydims, xdims = [int(s) for s in input_file.readline().split()]
+        # Our x points North, so the first thing we read is y, not x.
+        ymin, ymax = [float(s) for s in input_file.readline().split()]
+        xmin, xmax = [float(s) for s in input_file.readline().split()]
+        dmin, dmax = [float(s) for s in input_file.readline().split()]
+        field = np.fromiter((float(s)
+                             for line in input_file
+                             for s in line.split()),
+                            dtype=dtype)
```

Comment:
This call should be replaced with a `numpy.loadtxt`. Then we don't need to reshape the array before passing it to `DataArray`. It would also let us check the shape against `ydims`and `xdims`.

Reply:
This one I am having trouble with. When I read in .grd files `numpy.loadtxt` is having issues with line endings. I will keep working on this.

Reply:
I vaguely remember having similar issues. Let me know if you need any help with this.

Reply:
I haven't been able to figure out an easy way to parse through a `.grd` file with `np.loadtxt` because of the file formatting. It looks like it uses a `\n` newline character every 10 entries and then again at the end of each row with ` \n` with a space. It's fairly easy to count the number of ` \n` entries and get the number of rows, but there has to be a way to do it all in one pass with `np.loadtxt` or `np.genfromtxt`. Input is very welcome

Reply:
Right, I forgot about those blank lines that the surfer grid has. Probably best to not use `loadtxt` then. But you could still add a check that the number of elements read is consistent with `ydims*xdims`.

Reply:
I added a check that the dimensions from the header match the data